src/components/UpgradePlanForm/UpgradePlanForm.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import axios from 'axios';
+import { FormattedMessage } from 'react-intl';
+
+class UpgradePlanForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      userPlanModel: null,
+      formIsValid: false,
+    };
+
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  componentWillMount() {
+    const idUserType = this.props.isSubscriber ? 4 : 2;
+    axios
+      .get(
+        process.env.REACT_APP_API_URL +
+          '/SendUpgradePlanContactEmail/GetUpgradePlanData?idUserType=' +
+          idUserType,
+        {
+          withCredentials: true,
+        },
+      )
+      .then((response) => {
+        this.setState({ userPlanModel: response.data.data });
+      });
+  }
+
+  submitForm() {
+    //TODO research axios cancel request
+
+    fetch(process.env.REACT_APP_API_URL + '/SendUpgradePlanContactEmail/SendEmailUpgradePlan', {
+      method: 'post',
+      crossDomain: true,
+      body: JSON.stringify(this.state.userPlanModel),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      credentials: 'include',
+    })
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        this.props.handleClose();
```

Comment:
What do you think about using async/await?

Reply:
Why is better?

Reply:
It is easier to read, write, understand and less error-prone. Does it not?